test/unit/math/mix/core/operator_subtraction_test.cpp

```diff
@@ -6,3 +6,136 @@ TEST(mathMixCore, operatorSubtraction) {
   stan::test::expect_common_binary(f, disable_lhs_int);
   stan::test::expect_complex_common_binary(f);
 }
+
+TEST(mathMixCore, operatorSubtractionMatrixSmall) {
+  // This calls operator- under the hood
+  auto f = [](const auto& x1, const auto& x2) {
+    return stan::math::subtract(x1, x2);
+  };
+  stan::test::ad_tolerances tols;
+  tols.hessian_hessian_ = 1e-1;
+  tols.hessian_fvar_hessian_ = 1e-1;
+
+  double scalar_a = 10;
+  Eigen::VectorXd vector_v1(1);
+  vector_v1 << 3;
+  Eigen::RowVectorXd row_vector_rv1(1);
+  row_vector_rv1 << -2;
+  Eigen::MatrixXd matrix_m11(1, 1);
+  matrix_m11 << 1.5;
+  stan::test::expect_ad(tols, f, scalar_a, scalar_a);
+  stan::test::expect_ad(tols, f, scalar_a, vector_v1);
+  stan::test::expect_ad(tols, f, vector_v1, scalar_a);
+  stan::test::expect_ad(tols, f, scalar_a, row_vector_rv1);
+  stan::test::expect_ad(tols, f, row_vector_rv1, scalar_a);
+  stan::test::expect_ad(tols, f, scalar_a, matrix_m11);
+  stan::test::expect_ad(tols, f, matrix_m11, scalar_a);
+  stan::test::expect_ad(tols, f, matrix_m11, matrix_m11);
+
+  stan::test::expect_ad_matvar(tols, f, scalar_a, vector_v1);
+  stan::test::expect_ad_matvar(tols, f, vector_v1, scalar_a);
+  stan::test::expect_ad_matvar(tols, f, scalar_a, row_vector_rv1);
+  stan::test::expect_ad_matvar(tols, f, row_vector_rv1, scalar_a);
+  stan::test::expect_ad_matvar(tols, f, row_vector_rv1, vector_v1);
+  stan::test::expect_ad_matvar(tols, f, vector_v1, row_vector_rv1);
```

Comment:
These mixed type tests strike me as weird. I kinda want them gone cause they might be accidentally working now cause the linear sizes of the inputs are the same (and in reality I don't think we want to do the 1d vector ambiguity thing cause we don't do it at the language).

Reply:
I guess this is only checking that matvar behavior matches varmat, which is checking against prim, which is hopefully doing the right thing.

Reply:
Yeah, I mean I think it's okay to do here in math. As long as their is not a signature for it the language won't access it.